tools/themis/src/rules.rs

```diff
@@ -289,3 +296,41 @@ pub fn publishable_has_near_link(workspace: &Workspace) -> anyhow::Result<()> {
 
     Ok(())
 }
+
+/// Ensure all publishable crates do not depend on private crates
+pub fn recursively_publishable(workspace: &Workspace) -> anyhow::Result<()> {
+    let mut outliers = BTreeMap::new();
+    for pkg in workspace.members.iter().filter(|pkg| utils::is_publishable(pkg)) {
+        for dep in &pkg.parsed.dependencies {
+            if let Some(dep) = workspace.members.iter().find(|p| p.parsed.name == dep.name) {
+                if !utils::is_publishable(dep) {
+                    outliers
+                        .entry(dep.parsed.manifest_path.clone())
+                        .or_insert_with(Vec::new)
+                        .push(pkg.parsed.name.clone())
+                }
+            }
+        }
+    }
+    if !outliers.is_empty() {
+        bail!(ComplianceError {
+            msg: "These private packages are depended on by publishable packages".to_string(),
```

Comment:
The original message was hard to understand at the first glance, so I tried to rewrite it. What do you think?

```suggestion
            msg: "These private crates break publishable crates. Either make these private crates publishable or avoid using them in the publishable crates.".to_string(),
```

Reply:
Yeah, this makes sense. I, too, went through a couple of description iterations, the above happened to be the shortest with most clarity, but I agree that we don't need to optimize for length, so more descriptive (and in this case, suggestive) is good. If you can submit a PR to this effect, that'd be awesome.